src/query_router.rs

```diff
@@ -114,6 +115,47 @@ impl QueryRouter {
 
         let code = message_cursor.get_u8() as char;
 
+        // Check for any sharding regex matches in any queries
+        match code as char {
+            // For Parse and Query messages peek to see if they specify a shard_id as a comment early in the statement
+            'P' | 'Q' => {
+                if self.pool_settings.shard_id_regex.is_some()
+                    || self.pool_settings.sharding_key_regex.is_some()
+                {
+                    // Check only the first block of bytes configured by the pool settings
+                    let len = message_cursor.get_i32() as usize;
+                    let seg = cmp::min(len - 5, self.pool_settings.regex_search_limit);
+                    let initial_segment = String::from_utf8_lossy(&message_buffer[0..seg]);
+
+                    // Check for a shard_id included in the query
+                    if let Some(shard_id_regex) = &self.pool_settings.shard_id_regex {
+                        let shard_id = shard_id_regex.captures(&initial_segment).and_then(|cap| {
+                            cap.get(1).and_then(|id| id.as_str().parse::<usize>().ok())
+                        });
+                        if let Some(shard_id) = shard_id {
+                            debug!("Setting shard to {:?}", shard_id);
+                            self.set_shard(shard_id)
```

Comment:
I think you want to early return `None` here so the rest of the command parsing logic doesn't run for no reason.

Reply:
Ok, I think I misunderstood what this code is doing. The whole rest of the function is only about the special command parsing. I guess it's assumed that if you're passing the shard information in a comment then it won't need to look for any special commands? I'll put in the `return None` for these when it successfully finds sharding information in the new code.